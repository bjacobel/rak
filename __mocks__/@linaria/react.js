export const styled = jest.fn(strings => () => strings);
