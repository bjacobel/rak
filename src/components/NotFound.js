import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import styles from '../stylesheets/notFound.css';
import { link } from '../stylesheets/link.css';

export default class NotFound extends Component {
  render() {
    return (
      <div>
        <h1 className={styles['not-found']}>404: page not found</h1>
        <Link className={link} to="/">
          Home
        </Link>
      </div>
    );
  }
}
