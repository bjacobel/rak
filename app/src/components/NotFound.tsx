import { styled } from '@linaria/react';

import { center, openSans } from '../stylesheets/shared.css';
import StyledLink from './StyledLink';

const Header = styled.h1`
  ${center}
  ${openSans}
`;

export default () => (
  <div>
    <Header>404: page not found</Header>
    <StyledLink to="/">Home</StyledLink>
  </div>
);
