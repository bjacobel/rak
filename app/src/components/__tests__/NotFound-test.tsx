import { ReactNode } from 'react';
import { Route, Switch } from 'wouter';

import { render } from '../../testing/rtl';
import NotFound from '../NotFound';

describe('404 component', () => {
  let notFound: ReactNode;

  beforeEach(() => {
    notFound = (
      <Switch>
        <Route component={NotFound} path="*" />
      </Switch>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(notFound);

    expect(asFragment()).toMatchSnapshot();
  });

  it('displays 404 message', () => {
    const { getByText } = render(notFound);

    expect(getByText('404: page not found')).toBeInTheDocument();
  });

  it('has a link back to home', () => {
    const { getByRole } = render(notFound);
    const homeLink = getByRole('link');

    expect(homeLink).toBeInTheDocument();

    expect(new URL((homeLink as HTMLLinkElement).href).pathname).toEqual('/');
  });
});
