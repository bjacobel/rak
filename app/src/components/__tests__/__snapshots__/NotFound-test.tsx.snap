// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`404 component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      404: page not found
    </h1>
    <a
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;
